### ISSUE TITLE
Number of locations per minute greater than 10

### ISSUE THREAD

**Issue by asaddigital2809** [transistorsoft/flutter_background_geolocation#1396]
## Your Environment

Plugin version: flutter_background_geolocation: ^4.16.3

Platform: iOS or Android : IOS

OS version: 18

Device manufacturer / model: Apple

Flutter info (flutter doctor):
[!] Flutter (Channel [user-branch], 3.22.0, on macOS 13.6.9 22G830 darwin-x64, locale en-GB)
! Flutter version 3.22.0 on channel [user-branch] at /Users/naeemtariq/Desktop/Developer/flutter
Currently on an unknown channel. Run flutter channel to switch to an official channel.
If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
! Upstream repository unknown source is not a standard remote.
Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss this error.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.64.2)
[✓] VS Code (version 1.94.0)
[✓] Connected device (1 available)
[✓] Network resources

Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
    bg.BackgroundGeolocation.ready(bg.Config(
            url:  'Url', 
            desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
            // enableHeadless: true,
            // disableStopDetection: true,
            stopOnTerminate: false,
            startOnBoot: true,
            stopOnStationary: false,
            debug: false,
            foregroundService: true,
            heartbeatInterval: 20,
            stationaryRadius: Platform.isAndroid ? 0 : 25,
            distanceFilter: Platform.isAndroid ? 0 : 25,
            locationUpdateInterval: 11000,
            fastestLocationUpdateInterval: 11000,
            maxRecordsToPersist: 800,
            allowIdenticalLocations: true,
            enableHeadless: true,
            maxDaysToPersist: 1,
            pausesLocationUpdatesAutomatically: false,
            preventSuspend: true,
            activityType: bg.Config.ACTIVITY_TYPE_AUTOMOTIVE_NAVIGATION,
            locationTemplate:
                '{"driver_lat":<%= latitude %>,"driver_lan":<%= longitude %>,"event":"<%= event %>","driver_id":$driverId,"loc_time":"<%= timestamp %>","orders":${[]}}',
            headers: {
              "Content-Type": "application/json; charset=UTF-8",
              "Authorization": 'Bearer $token'
            },
            notificationPriority: Platform.isAndroid ? 0 : -2,
            logLevel: Platform.isAndroid
                ? bg.Config.LOG_LEVEL_VERBOSE
                : bg.Config.LOG_LEVEL_OFF))
        .then((bg.State state) {
      if (!state.enabled) {
        bg.BackgroundGeolocation.start();
      }
    });
```

## Actual Behavior

![WhatsApp Image 2024-12-24 at 22 20 53_e9393705](https://github.com/user-attachments/assets/584c142c-21b2-4a4a-bc3d-fbfa27b555c6)

**Comment by christocracy** [comment]
Do you have a question?

**Comment by asaddigital2809** [comment]
> Do you have a question?

Yes definitely. why there are more than 10 locations per minute. I have added the log details pic

**Comment by christocracy** [comment]
> why there are more than 10 locations per minute.

How many are you expecting?  There is no guarantee on "number of locations per minute".

**Comment by christocracy** [comment]
There are a number of ways you can get more locations than expected, eg: many changes in speed (eg: stuck in a traffic jam).

There is no way to guarantee a fixed number of locations / minute.  Increase your distanceFilter.

I have no idea what those devices were doing.  Walking?  car ride?  City driving vs highway?

The answer would be in the logs.  See API docs `.emailLog` and analyze the plugin logs.

**Comment by asaddigital2809** [comment]
Driving Car on Highway

> There are a number of ways you can get more locations than expected, eg: many changes in speed (eg: stuck in a traffic jam).
> 
> There is no way to guarantee a fixed number of locations / minute. Increase your distanceFilter.
> 
> I have no idea what those devices were doing. Walking? car ride? City driving vs highway?
> 
> The answer would be in the logs. See API docs `.emailLog` and analyze the plugin logs.
